pallets/base/src/lib.rs

```diff
@@ -71,4 +79,18 @@ pub fn ensure_opt_string_limited<T: Config>(s: Option<&[u8]>) -> DispatchResult
     }
 }
 
+/// Try to pre-increment the counter `seq` and return the next ID to use.
+pub fn try_next_id_pre<T: Config, I: CheckedInc + Clone>(seq: &mut I) -> Result<I, DispatchError> {
+    let id = seq.checked_inc().ok_or(Error::<T>::IdSeqOverflow)?;
+    *seq = id.clone();
+    Ok(id)
+}
```

Comment:
Why use ` + Clone` instead of ` + Copy`?  Is there a reason that `Copy` can't be used?

Reply:
Just being slightly more general, accepting more types.